### PR TITLE
Fixes #8309: Automatically run `adb reverse tcp:8081 tcp:8081` when st…

### LINKED ISSUE #8309: [JS] Automatically run `adb reverse tcp:8081 tcp:8081` when starting packager
This would make developing on a device easier:
http://facebook.github.io/react-native/docs/running-on-device-android.html

### PR DESCRIPTION
**Pull request by burgalon**
in `runAndroid.js` I renamed the `buildAndRun()` into `run()` and added a call to `runAdbReverse`

### CODE REVIEW

**Comment by facebook-github-bot-5** [comment]
By analyzing the blame information on this pull request, we identified @jimthedev and @jsdevel to be potential reviewers.

**Comment by facebook-github-bot-5** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

File: local-cli/runAndroid/runAndroid.js [hunk 1]
```diff
@@ -71,10 +71,30 @@ function checkAndroid(args) {
   return fs.existsSync(path.join(args.root, 'android/gradlew'));
 }
 
+function getAdbPath() {
+  return process.env.ANDROID_HOME
+    ? process.env.ANDROID_HOME + '/platform-tools/adb'
+    : 'adb';
+}
```
Comment (mkonicek) [inline local-cli/runAndroid/runAndroid.js]: Minor thing: can you add an empty line after `}` please?

**Comment by facebook-github-bot-9** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

File: local-cli/runAndroid/runAndroid.js [hunk 2]
```diff
@@ -71,10 +71,30 @@ function checkAndroid(args) {
   return fs.existsSync(path.join(args.root, 'android/gradlew'));
 }
 
+function getAdbPath() {
```
Comment (mkonicek) [inline local-cli/runAndroid/runAndroid.js]: Do we use `_` for private helper functions in this file? Should this be `_getAdbPath`?

**Comment by mkonicek** [comment]
This is amazing, thank you @burgalon!

File: local-cli/runAndroid/runAndroid.js (hunk 1 above)
Comment (burgalon) [inline local-cli/runAndroid/runAndroid.js]: Added :)

File: local-cli/runAndroid/runAndroid.js [hunk 3]
```diff
@@ -71,10 +71,38 @@ function checkAndroid(args) {
...
+  return process.env.ANDROID_HOME
+    ? process.env.ANDROID_HOME + '/platform-tools/adb'
+    : 'adb';
+}
+
+// Runs ADB reverse tcp:8081 tcp:8081 to allow loading the jsbundle from the packager
```
Comment (mkonicek) [inline local-cli/runAndroid/runAndroid.js]: Minor thing: Runs 'adb reverse tcp:8081 tcp:8081' to allow loading the JS bundle from the packager when using a physical device

- adb is usually spelled in lower case: https://developer.android.com/studio/command-line/adb.html
- JS bundle is the usual spelling
- we should say the purpose of this function is running on physical devices (on emulators it's not needed)

Comment (mkonicek) [inline local-cli/runAndroid/runAndroid.js]: We can fix that in a followup commit, let's shipit :)

**Comment by mkonicek** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-3** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1016168865146487/int_phab) to review.

**Comment by burgalon** [comment]
Fixes #8309

**Comment by mkonicek** [comment]
Reviewed in the Tel Aviv hackathon.

**Comment by himat** [comment]
Was this ever merged or no? What is the best way to auto run the adb reverse command now?

**Comment by caycecollins** [comment]
@himat, the cli tools have been moved to the [react-native-community/cli](https://github.com/react-native-community/cli) project. For completeness, you can check the `runAndroid` function, which includes the `tryRunAdbReverse` call [here](https://github.com/react-native-community/cli/blob/master/packages/platform-android/src/commands/runAndroid/index.ts#L232). Hope this helps!

**Pull request closed without merging**